运行motan-demo报错

运行motan-demo的motan-demo-client中的DemoRpcClient，spring装配，zookeeper注册中心，motan版本0.1.0，报错如下，

> Connected to the target VM, address: '127.0.0.1:49527', transport: 'socket'
Disconnected from the target VM, address: '127.0.0.1:49527', transport: 'socket'
Exception in thread "main" org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'motanDemoReferer': FactoryBean threw exception on object creation; nested exception is com.weibo.api.motan.exception.MotanFrameworkException: error_message: Create registry false for url:zookeeper://127.0.0.1:2181/com.weibo.api.motan.registry.RegistryService?group=default_rpc, status: 503, error_code: 20004,r=
	at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:175)
	at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.getObjectFromFactoryBean(FactoryBeanRegistrySupport.java:103)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getObjectForBeanInstance(AbstractBeanFactory.java:1585)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:254)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1051)
	at com.weibo.motan.demo.client.DemoRpcClient.main(DemoRpcClient.java:29)
Caused by: com.weibo.api.motan.exception.MotanFrameworkException: error_message: Create registry false for url:zookeeper://127.0.0.1:2181/com.weibo.api.motan.registry.RegistryService?group=default_rpc, status: 503, error_code: 20004,r=
	at com.weibo.api.motan.registry.support.AbstractRegistryFactory.getRegistry(AbstractRegistryFactory.java:63)
	at com.weibo.api.motan.cluster.support.ClusterSupport.getRegistry(ClusterSupport.java:129)
	at com.weibo.api.motan.cluster.support.ClusterSupport.init(ClusterSupport.java:89)
	at com.weibo.api.motan.config.handler.SimpleConfigHandler.buildClusterSupport(SimpleConfigHandler.java:52)
	at com.weibo.api.motan.config.RefererConfig.createClusterSupport(RefererConfig.java:184)
	at com.weibo.api.motan.config.RefererConfig.initRef(RefererConfig.java:128)
	at com.weibo.api.motan.config.RefererConfig.getRef(RefererConfig.java:85)
	at com.weibo.api.motan.config.springsupport.RefererConfigBean.getObject(RefererConfigBean.java:41)
	at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:168)
	... 6 more
Caused by: org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 200
	at org.I0Itec.zkclient.ZkClient.connect(ZkClient.java:880)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:98)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:92)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:80)
	at com.weibo.api.motan.registry.zookeeper.ZookeeperRegistryFactory.createRegistry(ZookeeperRegistryFactory.java:44)
	at com.weibo.api.motan.registry.support.AbstractRegistryFactory.getRegistry(AbstractRegistryFactory.java:56)
	... 14 more

但是，motan-demo-server没有报错，是不是我哪里配错了？或者是环境问题？

--- a/docs/wiki/zh_configuration.md
+++ b/docs/wiki/zh_configuration.md
@@ -53,6 +53,7 @@
 | regProtocol            | String  |         | 注册协议                     |
 | address                | String  |         | 注册中心地址                 |
 | port                   | int     | 0       | 注册中心缺省端口             |
+| connectTimeout         | int     | 1000    | 注册中心连接超时时间(毫秒)   |
 | requestTimeout         | int     | 200     | 注册中心请求超时时间(毫秒)   |
 | registrySessionTimeout | int     | 60s     | 注册中心会话超时时间(毫秒)   |
 | registryRetryPeriod    | int     | 30s     | 失败后重试的时间间隔         |

--- a/motan-core/src/main/java/com/weibo/api/motan/config/RegistryConfig.java
+++ b/motan-core/src/main/java/com/weibo/api/motan/config/RegistryConfig.java
@@ -46,6 +46,9 @@ public class RegistryConfig extends AbstractConfig {
     // 注册中心请求超时时间(毫秒)
     private Integer requestTimeout;
 
+    // 注册中心连接超时时间(毫秒)
+    private Integer connectTimeout;
+
     // 注册中心会话超时时间(毫秒)
     private Integer registrySessionTimeout;
 
@@ -166,6 +169,14 @@ public class RegistryConfig extends AbstractConfig {
 
     public Boolean isDefault() {
         return isDefault;
+    }
+
+    public Integer getConnectTimeout() {
+        return connectTimeout;
+    }
+
+    public void setConnectTimeout(Integer connectTimeout) {
+        this.connectTimeout = connectTimeout;
     }
 
     /**

--- a/motan-core/src/main/resources/META-INF/motan.xsd
+++ b/motan-core/src/main/resources/META-INF/motan.xsd
@@ -65,6 +65,11 @@
                     <xsd:documentation><![CDATA[ 注册中心缺省端口. ]]></xsd:documentation>
                 </xsd:annotation>
             </xsd:attribute>
+            <xsd:attribute name="connectTimeout" type="xsd:integer" use="optional">
+                <xsd:annotation>
+                    <xsd:documentation><![CDATA[ 注册中心连接超时时间(毫秒). ]]></xsd:documentation>
+                </xsd:annotation>
+            </xsd:attribute>
             <xsd:attribute name="requestTimeout" type="xsd:string" use="optional">
                 <xsd:annotation>
                     <xsd:documentation><![CDATA[ 注册中心请求超时时间(毫秒). ]]></xsd:documentation>

--- a/motan-demo/motan-demo-client/src/main/resources/motan_demo_client.xml
+++ b/motan-demo/motan-demo-client/src/main/resources/motan_demo_client.xml
@@ -23,7 +23,7 @@
     <!-- 注册中心配置 使用不同注册中心需要依赖对应的jar包。-->
     <!--<motan:registry regProtocol="local" name="registry"/>-->
     <!--<motan:registry regProtocol="consul" name="registry" address="127.0.0.1:8500"/>-->
-    <motan:registry regProtocol="zookeeper" name="registry" address="127.0.0.1:2181"/>
+    <motan:registry regProtocol="zookeeper" name="registry" address="127.0.0.1:2181" connectTimeout="2000"/>
 
     <!-- motan协议配置 -->
     <motan:protocol default="true" name="motan" haStrategy="failover"

--- a/motan-demo/motan-demo-client/src/main/resources/motan_demo_client_direct.xml
+++ b/motan-demo/motan-demo-client/src/main/resources/motan_demo_client_direct.xml
@@ -19,7 +19,7 @@
 	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:motan="http://api.weibo.com/schema/motan"
 	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.5.xsd
        http://api.weibo.com/schema/motan http://api.weibo.com/schema/motan.xsd">
-	<import resource="classpath*:spring/configloader.xml"/>
+
 	<!-- 注册中心配置 -->
 	<motan:registry regProtocol="local" name="motanDemo" />
 

--- a/motan-demo/motan-demo-server/src/main/resources/motan_demo_server.xml
+++ b/motan-demo/motan-demo-server/src/main/resources/motan_demo_server.xml
@@ -27,7 +27,7 @@
     <!-- 注册中心配置 使用不同注册中心需要依赖对应的jar包。如果不使用注册中心，可以把check属性改为false，忽略注册失败。-->
     <!--<motan:registry regProtocol="local" name="registry" />-->
     <!--<motan:registry regProtocol="consul" name="registry" address="127.0.0.1:8500"/>-->
-    <motan:registry regProtocol="zookeeper" name="registry" address="127.0.0.1:2181"/>
+    <motan:registry regProtocol="zookeeper" name="registry" address="127.0.0.1:2181" connectTimeout="2000"/>
 
     <!-- 协议配置。为防止多个业务配置冲突，推荐使用id表示具体协议。-->
     <motan:protocol id="demoMotan" default="true" name="motan"

--- a/motan-registry-zookeeper/src/main/java/com/weibo/api/motan/registry/zookeeper/ZookeeperRegistryFactory.java
+++ b/motan-registry-zookeeper/src/main/java/com/weibo/api/motan/registry/zookeeper/ZookeeperRegistryFactory.java
@@ -37,7 +37,7 @@ public class ZookeeperRegistryFactory extends AbstractRegistryFactory {
     @Override
     protected Registry createRegistry(URL registryUrl) {
         try {
-            int timeout = registryUrl.getIntParameter(URLParamType.requestTimeout.getName(), URLParamType.requestTimeout.getIntValue());
+            int timeout = registryUrl.getIntParameter(URLParamType.connectTimeout.getName(), URLParamType.connectTimeout.getIntValue());
             int sessionTimeout =
                     registryUrl.getIntParameter(URLParamType.registrySessionTimeout.getName(),
                             URLParamType.registrySessionTimeout.getIntValue());

